src/main/java/de/pierreschwang/spigotlib/inventory/SimplePaginatedInventory.java

```diff
@@ -105,22 +189,39 @@ public SimplePaginatedInventory addItem(ItemFactory<?> item) {
         return this.addItem(item.apply(), event -> {});
     }
 
-    public void open(int page, Player... players) {
-        if(page < 0) {
-            throw new PageIndexOutOfBoundsException("Page number can not be less than 1");
+    public void refresh(int newPage, Player player) {
+        if(newPage > getMaxPage() || newPage < 1) {
+            return;
         }
-        setCurrentPage(page);
-        int offset = getOffsetForPage(page);
-        for(int index = offset; index < offset + getInventory().getSize(); index++) {
-            getInventory().setItem(index - offset, getInventoryContents().get(index));
+        int offset = setCurrentPage(newPage).getOffsetForPage(newPage);
+        getDynamicSlots().forEach(dynamicSlot -> getInventory().setItem(dynamicSlot, getInventoryContents().get(dynamicSlot + offset)));
+        player.updateInventory();
+        setTitle(getInventory().getTitle().replace("%page%", String.valueOf(getCurrentPage())), player);
+    }
+
+    public void open(int page, Player player) {
+        if(page < 1) {
+            throw new PaginatedInventoryException("Page number (" + page +  ") can not be less than 1");
         }
-        for (Player player : players) {
-            player.openInventory(getInventory());
+        if(page > getMaxPage()) {
+            throw new PaginatedInventoryException("Page number (" + page + ") can not be greater than the maximum amount of pages (currently " + getMaxPage() + ")");
         }
+        player.openInventory(getInventory());
+        setTitle(getInventory().getTitle().replace("%page%", String.valueOf(page)), player);
```

Comment:
Insert check if inventory title contains placeholder to save Reflection calls if not necessary.